controllers/machineset_controller.go

```diff
@@ -534,8 +534,8 @@ func (r *MachineSetReconciler) MachineToMachineSets(o client.Object) []ctrl.Requ
 		}
 	}
 
-	mss := r.getMachineSetsForMachine(context.TODO(), m)
-	if len(mss) == 0 {
+	mss, err := r.getMachineSetsForMachine(context.TODO(), m)
+	if len(mss) == 0 || err != nil {
```

Comment:
I wonder if we should log something if we get an error here, because otherwise it never surfaces and it's really hard to debug.

Reply:
+1
And maybe separating the `err != nil` in a separate block, probably which would be done if the error is logged.